llvm/tools/obj2yaml/xcoff2yaml.cpp

```diff
@@ -131,7 +209,119 @@ Error XCOFFDumper::dumpSymbols() {
 
     Sym.Type = SymbolEntRef.getSymbolType();
     Sym.StorageClass = SymbolEntRef.getStorageClass();
-    Sym.NumberOfAuxEntries = SymbolEntRef.getNumberOfAuxEntries();
+    uint8_t NumOfAuxSym = SymbolEntRef.getNumberOfAuxEntries();
+    Sym.NumberOfAuxEntries = NumOfAuxSym;
+
+    if (NumOfAuxSym) {
+      std::vector<std::unique_ptr<XCOFFYAML::AuxSymbolEnt>> AuxEntTbl;
+      switch (Sym.StorageClass) {
+      case XCOFF::C_FILE: {
+        for (uint8_t I = 1; I <= NumOfAuxSym; ++I) {
```

Comment:
This loop is still inline in the switch when I asked for it to be pulled out into its own function. Please address all my comments or explicitly say you're either not doing them (and why not) or that you're going to do them in a different diff update. See https://llvm.org/docs/CodeReview.html#acknowledge-all-reviewer-feedback.

Reply:
I didn't fully catch your comments before and sorry about it. Now the whole case body is pulled out into a separate function instead of partial codes.